### PR TITLE
feat: add granular size constraints APIs

### LINKED ISSUE #7075: minWidth has no effect without minHeight in tauri windows config
### Describe the bug

When configuring the minWidth property in tauri's windows configuration, it does not take effect until a minHeight is defined as well. Here is an example.
image

### Reproduction

1. Define minWidth in the windows configuration (making sure minHeight is not set).
2. Obersve the window being resizable beyond the minimum value.
3. Define minHeight in the windows configuration, keeping minWidth untouched.
4. Observe minWidth taking effect and effectively constraining window resizing.

### Expected behavior

The window is constrained to not be resizable beyond the minWidth value, even if the minHeight value is not defined.

### Platform and versions

```text
[✔] Environment
    - OS: Windows 10.0.19045 X64
    ✔ WebView2: 113.0.1774.57
    ✔ MSVC:
        - Visual Studio Community�2019
        - Visual Studio Build Tools 2017
    ✔ rustc: 1.69.0 (84c898d65 2023-04-16)
    ✔ Cargo: 1.69.0 (6e9a83356 2023-04-12)
    ✔ rustup: 1.26.0 (5af9b9484 2023-04-05)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (default)
    - node: 16.13.1
    - npm: 9.6.6

[-] Packages
    - tauri [RUST]: 1.3.0
    - tauri-build [RUST]: 1.3.0
    - wry [RUST]: 0.24.3
    - tao [RUST]: 0.16.1
    - @tauri-apps/api [NPM]: 1.3.0
    - @tauri-apps/cli [NPM]: 1.3.1

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../dist
    - devPath: http://localhost:1420/
    - framework: Svelte
    - bundler: Vite
```

### Stack trace

_No response_

### Additional context

_No response_

### LINKED ISSUE #10227: [bug] The `minWidth` of the window is invalid.
### Describe the bug

The `minWidth` of the window is invalid.

### Reproduction

1. create a tauri project
2. set `minWidth` in `tauri.config.json`
3. run command `pnpm tauri dev`

### Expected behavior

_No response_

### Full `tauri info` output

```text
[✔] Environment
    - OS: Windows 10.0.26100 X64
    ✔ WebView2: 126.0.2592.87
    ✔ MSVC: Visual Studio Community 2022
    ✔ rustc: 1.79.0 (129f3b996 2024-06-10)
    ✔ cargo: 1.79.0 (ffa9cf99a 2024-06-03)
    ✔ rustup: 1.27.1 (54dd3d00f 2024-04-24)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (default)
    - node: 20.15.0
    - pnpm: 9.4.0
    - yarn: 1.22.22
    - npm: 10.7.0

[-] Packages
    - tauri [RUST]: 2.0.0-beta.23
    - tauri-build [RUST]: 2.0.0-beta.18
    - wry [RUST]: 0.41.0
    - tao [RUST]: 0.28.1
    - @tauri-apps/api [NPM]: 2.0.0-beta.14
    - @tauri-apps/cli [NPM]: 2.0.0-beta.21

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../dist
    - devUrl: http://localhost:1420/
    - framework: Vue.js
    - bundler: Vite
```

### Stack trace

_No response_

### Additional context

image

### PR DESCRIPTION
**Pull request by amrbashir**
closes #7075
closes #10227

### CODE REVIEW

**Review by FabianLars — DISMISSED** [review]

File: core/tauri-runtime-wry/src/lib.rs [hunk 1]
```diff
@@ -1144,6 +1164,7 @@ pub enum WindowMessage {
   SetSize(Size),
   SetMinSize(Option<Size>),
   SetMaxSize(Option<Size>),
+  SetSizConstraints(WindowSizeConstraints),
```
Comment (Legend-Master) [inline core/tauri-runtime-wry/src/lib.rs]:
```suggestion
  SetSizeConstraints(WindowSizeConstraints),
```

Comment (amrbashir) [reply core/tauri-runtime-wry/src/lib.rs]: nice catch

**Comment by Legend-Master** [comment]
Should we also expose this to js side?

**Comment by Legend-Master** [comment]
Testing with `"windows": [{"minWidth": 500,"width": 800,"label": "main1"}],` in the config file of api example, seems like the size of that window gets capped at 500px and can't be resized, can resize height though

**Comment by amrbashir** [comment]
> Should we also expose this to js side?

done

**Comment by amrbashir** [comment]
> Testing with `"windows": [{"minWidth": 500,"width": 800,"label": "main1"}],` in the config file of api example, seems like the size of that window gets capped at 500px and can't be resized, can resize height though

was a small typo, that set `max_width` with the value of `min_width`, it is now fixed

File: core/tauri-runtime-wry/src/lib.rs [hunk 2]
```diff
@@ -788,6 +788,7 @@ impl WindowBuilder for WindowBuilderWrapper {
       if let Some(max_height) = config.max_height {
         constraints.max_height = Some(ToaLogicalUnit::new(max_height).into());
       }
+      dbg!(&constraints);
```
Comment (FabianLars) [inline core/tauri-runtime-wry/src/lib.rs]:
```suggestion
```

**Comment by Legend-Master** [comment]
> was a small typo, that set max_width with the value of min_width, it is now fixed

> oh no, i'm also blind to typos now D:

🙃

File: tooling/api/src/window.ts [hunk 3]
```diff
@@ -1311,6 +1318,35 @@ class Window {
     })
   }
 
+  /**
+   * Sets the window inner size constraints.
+   * @example
+   * ```typescript
+   * import { getCurrent, PhysicalSize } from '@tauri-apps/api/window';
+   * await getCurrent().setSizeConstraints(new PhysicalSize(600, 500));
```
Comment (Legend-Master) [inline tooling/api/src/window.ts]: This doesn't seem to be correct

Comment (amrbashir) [inline tooling/api/src/window.ts]:
```suggestion
   * import { getCurrentWindow } from '@tauri-apps/api/window';
   * await getCurrentWindow().setSizeConstraints({ minWidth: 300 });
```

**Review by Legend-Master — APPROVED** [review]

**Review by FabianLars — APPROVED** [review]

**Pull request merged**